### PR TITLE
Hide project security setting tab from non-authorized users

### LINKED ISSUE #1477: Improve UI when non-accessible users access project security settings
Currently users that don't have access to the MFA page attached (assuming by role, this screenshot was created by an `Editor`) 

Possible solutions:
- Hide menu item for MFA on navigation and 404 if user tries to access it

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNjRUQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--146d90643b27ee3ca882b6aa5ff088e188760ddd/image.png)

### PR DESCRIPTION
**Pull request by midigofrank**
## Related issue

Fixes #1477

## Review checklist

- [x] I have performed a **self-review** of my code
- [x] I have verified that all appropriate **authorization policies** have been implemented and tested
- [x] If needed, I have updated the **changelog**
- [ ] Product has **QA'd** this feature

### CODE REVIEW

**Review by elias-ba — CHANGES_REQUESTED** [review]
Hey @midigofrank this is looking really great. I just could spot that when you enter the security tab URL in the browser (`http://localhost:4000/projects/4adf2644-ed4e-4f97-a24c-ab35b3cb1efa/settings#security`) it renders this blank page.

image

File: test/support/conn_case.ex [hunk 1]
```diff
@@ -132,24 +132,19 @@ defmodule LightningWeb.ConnCase do
   Setup helper that creates a user adds them as project user with a given role and logs them them in
   """
   def setup_project_user(conn, project, role) do
-    user = Lightning.AccountsFixtures.user_fixture()
+    user = Lightning.Factories.insert(:user)
```
Comment (jyeshe) [inline test/support/conn_case.ex:135]: great you've replaced! importing would look like more with the rest of the code

File: lib/lightning_web/live/project_live/settings.html.heex [hunk 2]
```diff
@@ -473,7 +474,10 @@
           </.table>
         </LightningWeb.Components.Common.panel_content>
 
-        <LightningWeb.Components.Common.panel_content for_hash="security">
+        <LightningWeb.Components.Common.panel_content
+          :if={can_edit_project(assigns)}
```
Comment (jyeshe) [inline lib/lightning_web/live/project_live/settings.html.heex:478]: :+1:

**Review by jyeshe — APPROVED** [review]
Even pasting the url it doesn't show the security option :clap: 
Small detail on the testing just to keep the same style with the rest of the code.

**Review by elias-ba — APPROVED** [review]
@midigofrank this is looking great. Nice job 🎉

**Review by taylordowns2000 — COMMENTED** [review]
@midigofrank , @jyeshe , i've merged this down but want to note two things:
1. there's a commented out console.log that should be removed
2. i don't think this is the quickest fix for the desired UX. we want non admin users to be able to view this interface but for the MFA toggle to be `disabled`. they can _see_ whether it's enabled or not, but not modify it. is that harder to do than hiding the entire page? (hiding the page is _not_ what we want from a UX perspective.)

**Comment by jyeshe** [comment]
> @midigofrank , @jyeshe , i've merged this down but want to note two things:
> 
> 1. there's a commented out console.log that should be removed
> 2. i don't think this is the quickest fix for the desired UX. we want non admin users to be able to view this interface but for the MFA toggle to be `disabled`. they can _see_ whether it's enabled or not, but not modify it. is that harder to do than hiding the entire page? (hiding the page is _not_ what we want from a UX perspective.) do we have some patterns elsewhere in the app where we make certain buttons or inputs `disabled`?

Hi @taylordowns2000 it was suggested to be hidden on the issue #1477 and I would choose this customization for non-admin as we are preventing to show options to users that they cannot/should not operate.

**Comment by taylordowns2000** [comment]
Ah, but we need to actually *SHOW* this stuff to folks who can’t modify it.
No sweat, just a miscommunication. Follow up with Nick in the new years if
you’ve got questions. We’ll open a new issue!

Taylor Downs
CEO, OpenFn <https://www.openfn.org>

**Pull request merged**